src/io/flutter/run/daemon/DevToolsService.java

```diff
@@ -92,23 +100,21 @@ private void setUpWithDaemon() {
           return;
         }
         if (address == null) {
-          Exception error = new Exception("DevTools address was null");
-          LOG.error(error);
-          devToolsInstance.completeExceptionally(error);
+          logExceptionAndComplete("DevTools address was null");
         }
         else {
           devToolsInstance.complete(new DevToolsInstance(address.host, address.port));
         }
       });
     }
     catch (ExecutionException e) {
-      e.printStackTrace();
+      logExceptionAndComplete(e);
     }
 
     ProjectManager.getInstance().addProjectManagerListener(project, new ProjectManagerListener() {
       @Override
       public void projectClosing(@NotNull Project project) {
-        devToolsInstance = new CompletableFuture<>();
+        devToolsInstance = null;
```

Comment:
Do we need to dispose() the devToolsInstance here?

Reply:
Is dispose mainly for if an object has dependencies? Since the instance just holds a hostname and port, I think it may not be needed.

Reply:
Gotcha (yeah, it's just needed for heavier weight things).